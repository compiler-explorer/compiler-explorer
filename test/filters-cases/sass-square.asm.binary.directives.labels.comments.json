{
  "asm": [
    {
      "text": "_Z6squarePii:",
      "source": null,
      "labels": []
    },
    {
      "opcodes": [
        "e4",
        "5d",
        "00",
        "10",
        "01",
        "40",
        "00",
        "28"
      ],
      "address": 8,
      "text": " MOV R1, c[0x0][0x44]",
      "source": {
        "file": null,
        "line": 3
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "04",
        "1c",
        "00",
        "94",
        "00",
        "00",
        "00",
        "2c"
      ],
      "address": 16,
      "text": " S2R R0, SR_CTAID.X",
      "source": {
        "file": null,
        "line": 3
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "23",
        "dc",
        "01",
        "20",
        "05",
        "40",
        "0e",
        "1b"
      ],
      "address": 24,
      "text": " ISETP.GE.AND P0, PT, R0, c[0x0][0x148], PT",
      "source": {
        "file": null,
        "line": 4
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "e7",
        "01",
        "00",
        "00",
        "00",
        "00",
        "00",
        "80"
      ],
      "address": 32,
      "text": " @P0 EXIT",
      "source": {
        "file": null,
        "line": 4
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "43",
        "9c",
        "00",
        "00",
        "05",
        "40",
        "01",
        "40"
      ],
      "address": 40,
      "text": " ISCADD R2.CC, R0, c[0x0][0x140], 0x2",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "e2",
        "dd",
        "00",
        "10",
        "00",
        "00",
        "00",
        "18"
      ],
      "address": 48,
      "text": " MOV32I R3, 0x4",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "e3",
        "dc",
        "00",
        "10",
        "05",
        "80",
        "86",
        "20"
      ],
      "address": 56,
      "text": " IMAD.HI.X R3, R0, R3, c[0x0][0x144]",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "85",
        "1c",
        "20",
        "00",
        "00",
        "00",
        "00",
        "84"
      ],
      "address": 72,
      "text": " LD.E R0, [R2]",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "a3",
        "1c",
        "01",
        "00",
        "00",
        "00",
        "00",
        "50"
      ],
      "address": 80,
      "text": " IMUL R4, R0, R0",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "85",
        "1c",
        "21",
        "00",
        "00",
        "00",
        "00",
        "94"
      ],
      "address": 88,
      "text": " ST.E [R2], R4",
      "source": {
        "file": null,
        "line": 5
      },
      "links": null,
      "labels": []
    },
    {
      "opcodes": [
        "e7",
        "1d",
        "00",
        "00",
        "00",
        "00",
        "00",
        "80"
      ],
      "address": 96,
      "text": " EXIT",
      "source": {
        "file": null,
        "line": 6
      },
      "links": null,
      "labels": []
    },
    {
      "text": ".L_1:",
      "source": null,
      "labels": []
    },
    {
      "opcodes": [
        "e7",
        "1d",
        "00",
        "e0",
        "ff",
        "ff",
        "03",
        "40"
      ],
      "address": 104,
      "text": " BRA `(.L_1)",
      "source": {
        "file": null,
        "line": 6
      },
      "links": null,
      "labels": []
    },
    {
      "text": ".L_18:",
      "source": null,
      "labels": []
    }
  ],
  "labelDefinitions": {
    "_Z6squarePii": 1,
    ".L_1": 13,
    ".L_18": 15
  }
}