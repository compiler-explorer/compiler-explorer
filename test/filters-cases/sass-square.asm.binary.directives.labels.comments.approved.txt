{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "_Z6squarePii:"
    },
    {
      "address": 8,
      "labels": [],
      "opcodes": [
        "2800400110005de4"
      ],
      "source": {
        "file": null,
        "line": 3,
        "mainsource": true
      },
      "text": " MOV R1, c[0x0][0x44]"
    },
    {
      "address": 16,
      "labels": [],
      "opcodes": [
        "2c00000094001c04"
      ],
      "source": {
        "file": null,
        "line": 3,
        "mainsource": true
      },
      "text": " S2R R0, SR_CTAID.X"
    },
    {
      "address": 24,
      "labels": [],
      "opcodes": [
        "1b0e40052001dc23"
      ],
      "source": {
        "file": null,
        "line": 4,
        "mainsource": true
      },
      "text": " ISETP.GE.AND P0, PT, R0, c[0x0][0x148], PT"
    },
    {
      "address": 32,
      "labels": [],
      "opcodes": [
        "80000000000001e7"
      ],
      "source": {
        "file": null,
        "line": 4,
        "mainsource": true
      },
      "text": " @P0 EXIT"
    },
    {
      "address": 40,
      "labels": [],
      "opcodes": [
        "4001400500009c43"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " ISCADD R2.CC, R0, c[0x0][0x140], 0x2"
    },
    {
      "address": 48,
      "labels": [],
      "opcodes": [
        "180000001000dde2"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " MOV32I R3, 0x4"
    },
    {
      "address": 56,
      "labels": [],
      "opcodes": [
        "208680051000dce3"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " IMAD.HI.X R3, R0, R3, c[0x0][0x144]"
    },
    {
      "address": 72,
      "labels": [],
      "opcodes": [
        "8400000000201c85"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " LD.E R0, [R2]"
    },
    {
      "address": 80,
      "labels": [],
      "opcodes": [
        "5000000000011ca3"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " IMUL R4, R0, R0"
    },
    {
      "address": 88,
      "labels": [],
      "opcodes": [
        "9400000000211c85"
      ],
      "source": {
        "file": null,
        "line": 5,
        "mainsource": true
      },
      "text": " ST.E [R2], R4"
    },
    {
      "address": 96,
      "labels": [],
      "opcodes": [
        "8000000000001de7"
      ],
      "source": {
        "file": null,
        "line": 6,
        "mainsource": true
      },
      "text": " EXIT"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_1:"
    },
    {
      "address": 104,
      "labels": [],
      "opcodes": [
        "4003ffffe0001de7"
      ],
      "source": {
        "file": null,
        "line": 6,
        "mainsource": true
      },
      "text": " BRA `(.L_1)"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_18:"
    }
  ],
  "labelDefinitions": {
    ".L_1": 13,
    ".L_18": 15,
    "_Z6squarePii": 1
  }
}
